tests/devices/qutrit_mixed/test_qutrit_mixed_apply_operation.py

```diff
@@ -320,3 +338,228 @@ def test_broadcasted_state(self, ml_framework, wire, two_qutrit_batched_state):
+        expected_derivative = np.diagonal(expected_derivative_state)
+        print(g, "\n")
+        print(expected_derivative)
+        assert qml.math.allclose(g, expected_derivative)
+
+    @pytest.mark.autograd
+    def test_trx_grad_autograd(self, two_qutrit_state, subspace):
+        """Test that the application of a trx gate is differentiable with autograd."""
+
+        state = qml.numpy.array(two_qutrit_state)
+
+        def f(phi):
+            op = qml.TRX(phi, wires=0, subspace=subspace)
+            new_state = apply_operation(op, state)
+            return measure(qml.probs(), new_state)
+
+        phi = qml.numpy.array(self.phi, requires_grad=True)
+
+        probs = f(phi)
+        g = qml.jacobian(lambda x: qml.math.real(f(x)))(phi)
+        self.compare_expected_result(phi, state, probs, subspace, g)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("use_jit", (True, False))
+    def test_trx_grad_jax(self, use_jit, two_qutrit_state, subspace):
+        """Test that the application of a trx gate is differentiable with jax."""
+
+        import jax
+
+        state = jax.numpy.array(two_qutrit_state)
+
+        def f(phi):
+            op = qml.TRX(phi, wires=0, subspace=subspace)
+            new_state = apply_operation(op, state)
+            return measure(qml.probs(), new_state)
+
+        if use_jit:
+            f = jax.jit(f)
+
+        phi = self.phi
+
+        probs = f(phi)
+        g = jax.jacobian(f)(phi)
+        self.compare_expected_result(phi, state, probs, subspace, g)
+
+    @pytest.mark.torch
+    def test_trx_grad_torch(self, two_qutrit_state, subspace):
+        """Tests the application and differentiation of a trx gate with torch."""
+
+        import torch
+
+        state = torch.tensor(two_qutrit_state)
+
+        def f(phi):
+            op = qml.TRX(phi, wires=0, subspace=subspace)
+            new_state = apply_operation(op, state)
+            return measure(qml.probs(), new_state)
+
+        phi = torch.tensor(self.phi, requires_grad=True)
+
+        probs = f(phi)
+        g = torch.autograd.functional.jacobian(f, phi)
+
+        self.compare_expected_result(
+            phi.detach().numpy(),
+            state.detach().numpy(),
+            probs.detach().numpy(),
+            subspace,
+            g.detach().numpy(),
+        )
+
+    @pytest.mark.tf
+    def test_trx_grad_tf(self, two_qutrit_state, subspace):
+        """Tests the application and differentiation of a trx gate with tensorflow"""
+        import tensorflow as tf
+
+        state = tf.Variable(two_qutrit_state)
+        phi = tf.Variable(0.8589, trainable=True)
+
+        with tf.GradientTape() as grad_tape:
+            op = qml.TRX(phi, wires=0, subspace=subspace)
+            new_state = apply_operation(op, state)
+            probs = measure(qml.probs(), new_state)
+
+        grads = grad_tape.jacobian(probs, [phi])
+        # tf takes gradient with respect to conj(z), so we need to conj the gradient
```

Comment:
That seems like it could cause a lot of problems... is that just in this example, or something more widespread?

Reply:
Sorry that was an old fragment I forgot to delete.